backend/remote-state/http/backend.go

```
@@ -0,0 +1,334 @@
+		b.username = v.(string)
+	}
+
+	if v, ok := data.GetOk("password"); ok {
+		b.password = v.(string)
+	}
+
+	client := &http.Client{}
+
+	if v, ok := data.GetOk("skip_cert_verification"); ok {
+		b.skipTLS = v.(bool)
+		if b.skipTLS {
+			// If skip_cert_verification = true, the address must be of type HTTPS
+			if !isHTTPS(addressURL) {
+				return fmt.Errorf("Address must be of type HTTPS if skip_cert_verification = true")
+			}
+			// If local_cert_ca_file is also set, raise an error
+			if data.Get("local_cert_ca_file").(string) != "" {
+				return fmt.Errorf("skip_cert_verification is true and local_cert_ca_file is set. please choose one or the other")
+			}
+			// If mutual_tls_authentication is also set, raise an error
+			if data.Get("mutual_tls_authentication").(bool) == true {
+				return fmt.Errorf("skip_cert_verification is true and mutual_tls_authentication is set. please choose one or the other")
+			} //else {
+			// Replace the client with one that ignores TLS verification
+			client = &http.Client{
+				Timeout: time.Second * 10,
+				Transport: &http.Transport{
+					TLSClientConfig: &tls.Config{
+						InsecureSkipVerify: true,
+					},
+				},
+			}
+		}
+	}
+
+	if v, ok := data.GetOk("local_cert_ca_file"); ok {
+		// If local_cert_ca_file exists, the address must be of type HTTPS
+		if !isHTTPS(addressURL) {
+			return fmt.Errorf("Address must be of type HTTPS if local_cert_ca_file is set")
+		} //else {
+		if data.Get("mutual_tls_authentication").(bool) == true {
+			return fmt.Errorf("mutual_tls_authentication is true and local_cert_ca_file is set. Please choose one or the other")
+		}
+		if data.Get("skip_cert_verification").(bool) == true {
+			return fmt.Errorf("skip_cert_verification is true and local_cert_ca_file is set. Please choose one or the other")
+		}
+		b.localCertCAFile = v.(string)
+
+		// Get the SystemCertPool, continue with an empty pool on error
+		rootCAs, _ := x509.SystemCertPool()
```

why do you ignore/suppress the error here? I think you'd want to halt and return error if one was returned by this call...

---

continue with an empty pool on error? We don't really care it was an error, we will create a brand new empty pool.
That's how I saw it was implemented by different people...
https://gist.github.com/olekukonko/6e864ba3ba84f19070a03627fe8052a5
https://forfuncsake.github.io/post/2017/08/trust-extra-ca-cert-in-go-app/